### PR TITLE
Support ByteDance-Seed/BAGEL-7B-MoT quantization in w4a16 format

### PR DESCRIPTION
**Pull request by lvliang-intel**
## Description

This PR adds proper BAGEL model quantization support to the standard AutoRound LLM quantization flow and fixes the exported quantization metadata required by downstream vLLM-Omni loading.

Main changes:

(1) Route BAGEL through the LLM compressor.
(2) Load BAGEL with a dedicated custom loader because transformers does not natively recognize the bagel architecture.
(3) Gracefully handle AutoConfig.from_pretrained failures for unsupported model types such as bagel.
(4) Export the correct block_name_to_quantize metadata so downstream runtimes only quantize BAGEL LLM blocks instead of non-LLM modules like connector or vision components.
(5) Add a BAGEL-specific ignore policy to preserve image-generation-sensitive modules in FP16:
     a. all moe_gen modules
     b. shared attention projections (q_proj, k_proj, v_proj, o_proj)
(6) Fix `save_pretrained()` in the BAGEL custom loader to use `state_dict()` instead of `named_parameters()`, ensuring registered buffers (e.g., rotary embedding caches) are included in the saved `model.safetensors` for correct reload and inference.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Other (please specify):

## Related Issues

https://github.com/intel/auto-round/issues/1608

## Checklist Before Submitting

- [x] My code has been tested locally.
- [ ] Documentation has been updated as needed.
- [ ] New or updated tests are included where applicable.

---

🔒 GitHub Advanced Security automatically protects Copilot coding agent pull requests. You can protect all pull requests by enabling Advanced Security for your repositories. [Learn more about Advanced Security.](https://gh.io/cca-advanced-security)

### CODE REVIEW

File: auto_round/utils/bagel_loader.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        for name, param in self.named_parameters():
+            state_dict[name] = param.data.contiguous()
+
+        # Remap weight names to match original BAGEL checkpoint format
+        # The BagelPipeline expects top-level names like:
+        #   language_model.model.layers.0.self_attn.q_proj.weight
+        #   connector.fc1.weight
+        #   vit_model.vision_model.embeddings...
+        #   encoder.*, decoder.* (VAE, but those are in ae.safetensors)
+        save_file(state_dict, os.path.join(output_dir, "model.safetensors"))
```
[the remark this answers is not in the document]
Comment (lvliang-intel) [reply auto_round/utils/bagel_loader.py]: @copilot apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1633#discussion_r3000844401)

File: auto_round/utils/bagel_loader.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        device_map: Device map for model placement.
+
+    Returns:
+        Tuple of (model, tokenizer).
+    """
+    # Load configs
+    config_path = os.path.join(model_path, "config.json")
+    with open(config_path, "r", encoding="utf-8") as f:
+        bagel_config_dict = json.load(f)
+
```
[the remark this answers is not in the document]
Comment (lvliang-intel) [reply auto_round/utils/bagel_loader.py:268]: @copilot+claude-sonnet-4.6 apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1633#discussion_r3000844429)

File: auto_round/utils/model.py [hunk 3]
```diff
@@ -331,6 +331,26 @@ def llm_load_model(
...
+        with open(_config_path) as _f:
+            _mt = json.load(_f).get("model_type")
+        if _mt == "bagel":
+            from auto_round.utils.bagel_loader import load_bagel_model
+
+            model, tokenizer = load_bagel_model(
+                pretrained_model_name_or_path,
+                torch_dtype=torch_dtype,
+            )
+            model = _to_model_dtype(model, model_dtype)
```
[the remark this answers is not in the document]
Comment (lvliang-intel) [reply auto_round/utils/model.py:355]: @copilot+claude-sonnet-4.6 apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1633#discussion_r3000844481)

[the remark this answers is not in the document]
Comment (lvliang-intel) [reply auto_round/utils/model.py:355]: @copilot+claude-sonnet-4.6 apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1633#discussion_r3000844500)

File: auto_round/utils/model.py [hunk 4]
```diff
@@ -750,10 +779,28 @@ def is_pure_text_model(model):
...
+        _model_type = getattr(model_or_path.config, "model_type", None)
+    elif isinstance(model_path, str) and os.path.isdir(model_path):
+        _cfg_path = os.path.join(model_path, "config.json")
+        if os.path.exists(_cfg_path):
+            with open(_cfg_path) as _f:
+                _model_type = json.load(_f).get("model_type")
+    if _model_type in _LLM_ONLY_MODEL_TYPES:
+        return False
     # For dummy model, model_path could be "".
     if model_path and not os.path.isdir(model_path):
```
[the remark this answers is not in the document]
Comment (lvliang-intel) [reply auto_round/utils/model.py]: @copilot+claude-sonnet-4.6 apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1633#discussion_r3000844527)

File: auto_round/utils/bagel_loader.py [hunk 5]
```diff
@@ -0,0 +1,362 @@
+# Copyright (c) 2025 Intel Corporation
```
Comment (yiliu30) [inline auto_round/utils/bagel_loader.py]: ### 
```suggestion
# Copyright (c) 2026 Intel Corporation
```

File: auto_round/utils/bagel_loader.py [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+                del config_dict[key]
+        config_dict["architectures"] = ["BagelForConditionalGeneration"]
+        config_dict["model_type"] = "bagel"
+        config_path = os.path.join(output_dir, "config.json")
+        with open(config_path, "w", encoding="utf-8") as f:
+            json.dump(config_dict, f, indent=2, ensure_ascii=False)
+
+        # Save all model parameters as safetensors
+        state_dict = {}
+        for name, param in self.named_parameters():
```
Comment (yiliu30) [inline auto_round/utils/bagel_loader.py:239]: Please handle buffers as well.

**Review by yiliu30 — APPROVED** [review]
LGTM. Please add some UTs.

**Comment by lvliang-intel** [comment]
> LGTM. Please add some UTs.

ut added
